launch/config.py

```diff
@@ -0,0 +1,385 @@
+import copy
+import os
+
+import yaml
+
+import launch.util
+import test_util.aws
+from dcos_installer.config import config_sample
+from gen import get_dcosconfig_source_target_and_templates, stringify_configuration
+from gen.build_deploy.bash import onprem_source
+from gen.calc import validate_true_false
+from gen.internals import resolve_configuration, Scope, Source, Target, validate_one_of
+from pkgpanda.util import load_string, load_yaml
+from ssh.validate import source as ssh_source
+
+
+def expand_path(path: str, config_dir: str):
+    """ Returns absolute paths by performing ~ expansion or prepending the
+    config_dir to make an absolute path
+    """
+    if path.startswith('/'):
+        # nothing to do, return the path
+        return path
+    if path.startswith('~'):
+        # Try this and raise LauncherError
+        home_dir = os.environ['HOME']
+        return path.join(home_dir, path[1:])
+    # path is relative to config_dir
+    return os.path.join(config_dir, path)
+
+
+def get_validated_config(config_path: str):
+    """ Takes a dcos-launch configuration file and:
+    - string-ifies nested dicts
+    - expands paths into absolute paths, supporting tilde expansion
+    - converts provider_info_filename into provider_info
+    """
+    config_dir = os.path.dirname(os.path.abspath(config_path))
+    # Try and catch YamlParseError
```

Comment:
Will add for the express purpose of providing a cleaner message when a garbled config is provided